File: java17/src/main/java/folio/codinginterview/application/repository/AccountRepository.java
package folio.codinginterview.application.repository;

import folio.codinginterview.domain.Account;
import folio.codinginterview.domain.UserId;

import java.util.Optional;
import java.util.concurrent.CompletableFuture;

/**
 * 口座管理リポジトリ。
 */
public interface AccountRepository {
    CompletableFuture<Optional<Account>> find(UserId userId);

    CompletableFuture<Void> upsert(UserId userId, Account account);

    CompletableFuture<Boolean> exists(UserId userId);
}

File: php/src/CodingInterview/Application/Repository/AccountRepository.php
<?php

declare(strict_types=1);

namespace Folio\CodingInterview\Application\Repository;

use Folio\CodingInterview\Domain\Account;
use Folio\CodingInterview\Domain\UserId;

/**
 * 口座管理リポジトリ。
 */
interface AccountRepository
{
    public function find(UserId $userId): ?Account;

    public function upsert(UserId $userId, Account $account): void;

    public function exists(UserId $userId): bool;
}
